sycl/source/detail/pi.cpp

```diff
@@ -386,8 +690,25 @@ static void initializePlugins(std::vector<plugin> *Plugins) {
       std::cerr << "SYCL_PI_TRACE[basic]: "
                 << "Plugin found and successfully loaded: "
                 << PluginNames[I].first << std::endl;
-  }
 
+    fillPlatformAndDeviceCache(Plugins->back());
```

Comment:
Could you please clarify why the cache is populated with the last plugin only?

Reply:
"Plugins" is the vector<plugin> that is incrementally filled in each iteration of the for loop.
The last plugin is the one that is just added at each iteration.
This means each plugin is populated. Not just only the last plugin of the vector.